### PR TITLE
Count reactor stall

### PR DESCRIPTION
**Pull request by aleksbykov**
For performance tests latency-with-operation it is required to
count number of reactor stalls operations per nemesis

If reactor stall event happened during operation/nemesis, it will be added to file
under dir with Operation/Nemesis name, if reactor stall event between operations,
it will be added to file under dir with name SteadyState

Then reactor stalls will be counted for each operations

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by aleksbykov** [comment]
Example of report:
Screenshot from 2023-06-22 18-13-15

**Comment by fgelcer** [comment]
> Example of report: Screenshot from 2023-06-22 18-13-15

few questions..
1. is this "per nemesis" not taking into consideration the repetitions, like grow cluster running 3 times `add_node` operation in it, right?
2. what does `less or equal` mean?
3. does this report have the decoded backtraces attached to it?

File: sdcm/reactor_stall_decoder.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+from pathlib import Path
+from collections import defaultdict
+
+
+LOGGER = logging.getLogger(__name__)
+DECODED_STALLS_DIR_NAME = "operations_reactor_stalls"
+REACTOR_STALL_MS_REGEXP = re.compile(r" (\d+) ms")
+
+
+class ReactorStallExtractor:  # pylint: disable=too-few-public-methods
```
Comment (soyacz) [inline sdcm/reactor_stall_decoder.py:14]: please add few comment lines what `ReactorStallExtractor` is.

File: sdcm/reactor_stall_decoder.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+                    continue
+                with open(node_file, 'w', encoding='utf-8') as encoded_stall_file:
+                    encoded_stall_file.write('\n'.join(node_data[node]))
+
+    def extract_reactor_stall_by_nemesis(self):
+        self._extract_reactor_stalls_events_by_nemesis()
+        self._save_per_operation_by_node()
+
+
+class ReactorStallCounter:  # pylint: disable=too-few-public-methods
```
Comment (soyacz) [inline sdcm/reactor_stall_decoder.py:87]: please add few comment lines what `ReactorStallCounter` is.

File: sdcm/reactor_stall_decoder.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        self.store_dir = base_dir.joinpath(DECODED_STALLS_DIR_NAME)
+        self.reactor_stalls_per_operation = defaultdict(dict)
+
+    @staticmethod
+    def _convert_str_to_path(value: str | Path) -> Path:
+        if isinstance(value, str):
+            return Path(value)
+        return value
+
+    def _extract_reactor_stalls_events_by_nemesis(self):
```
Comment (soyacz) [inline sdcm/reactor_stall_decoder.py:34]: please add bit of comment what this function does - it's bit complex.

**Comment by aleksbykov** [comment]
Close this PR, because implement another functionality, which allow count and gatther stats for events during run:
https://github.com/scylladb/scylla-cluster-tests/pull/6302

**Pull request closed without merging**